### PR TITLE
Greeting screen with terms and conditions and privacy policy

### PR DESCRIPTION
**Pull request by MarcRubio**

### CODE REVIEW

File: app/src/main/java/com/radixdlt/android/apps/wallet/ui/fragment/greeting/GreetingFragment.kt [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    }
+
+    /**
+     * View Margins for getStartedButton and radixLogo are set dynamically due to displaying this
+     * view in full screen including the status bar. Different phones have different size status
+     * bars which need to be accounted for.
+     * */
+    private fun setViewMargins() {
+        val paramsButton = greetingGetStartedButton.layoutParams as ConstraintLayout.LayoutParams
+        paramsButton.bottomMargin = getNavigationBarHeight() + ctx.dip(16)
```
Comment (Sajjon) [inline app/src/main/java/com/radixdlt/android/apps/wallet/ui/fragment/greeting/GreetingFragment.kt]: move `16` out to a local variable? :)

File: app/src/main/java/com/radixdlt/android/apps/wallet/ui/fragment/greeting/GreetingViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    private val _enableGetStartedButton = MutableLiveData<Boolean>()
+    val enableGetStartedButton: LiveData<Boolean> get() = _enableGetStartedButton
+
+    init {
+        _enableGetStartedButton.value = false
+    }
+
+    fun onTermsAndConditionsCheckedChange(isChecked: Boolean) {
+        termsAndConditionsAccepted = isChecked
+        _enableGetStartedButton.value = termsAndConditionsAccepted && privacyPolicyAccepted
```
Comment (Sajjon) [inline app/src/main/java/com/radixdlt/android/apps/wallet/ui/fragment/greeting/GreetingViewModel.kt]: create a computed property for `areBothTerms&ConditionsAndPrivacyPolicyAccepted` ?

File: app/src/main/res/layout/activity_enter_mnemonic.xml [hunk 3]
```diff
@@ -12,35 +12,21 @@
...
-        <com.google.android.material.textview.MaterialTextView
-            android:id="@+id/welcomeTextView"
-            android:layout_width="wrap_content"
+        <com.google.android.material.button.MaterialButton
+            android:id="@+id/createWalletFromMnemonicButton"
+            android:layout_width="0dp"
             android:layout_height="wrap_content"
             android:layout_marginStart="16dp"
-            android:layout_marginTop="24dp"
+            android:layout_marginTop="8dp"
```
Comment (Sajjon) [inline app/src/main/res/layout/activity_enter_mnemonic.xml:20]: how does this constant play with various size classes?

File: app/src/main/res/layout/activity_enter_mnemonic.xml [hunk 4]
```diff
@@ -67,6 +53,21 @@
...
         </com.google.android.material.textfield.TextInputLayout>
 
+        <com.google.android.material.textview.MaterialTextView
+            android:id="@+id/welcomeTextView"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="16dp"
+            android:layout_marginTop="24dp"
+            android:layout_marginEnd="16dp"
+            android:text="Create wallet"
```
Comment (Sajjon) [inline app/src/main/res/layout/activity_enter_mnemonic.xml:63]: Hard coded string? `Create wallet`? Or is it some "hint"?

File: app/src/main/res/layout/activity_enter_mnemonic.xml [hunk 5]
```diff
@@ -81,20 +82,20 @@
...
             android:layout_height="wrap_content"
             android:layout_marginStart="16dp"
-            android:layout_marginTop="8dp"
-            android:layout_marginEnd="16dp"
-            android:text="Create wallet from mnemonic"
-            android:textColor="@android:color/white"
-            app:backgroundTint="@color/colorAccent"
+            android:layout_marginTop="16dp"
+            android:layout_marginEnd="8dp"
+            android:text="Enter your 12 word mnemonic or seed"
```
Comment (Sajjon) [inline app/src/main/res/layout/activity_enter_mnemonic.xml:92]: Hard coded string?

File: app/src/main/res/layout/activity_enter_mnemonic.xml (hunk 4 above)
Comment (MarcRubio) [reply app/src/main/res/layout/activity_enter_mnemonic.xml:63]: This screen is old and will be gone/redone

**Review by Sajjon — APPROVED** [review]
Looks good, try to see if you can make the test we talked about a bit more stable, failed 4/5 times.

Left some comments.

File: app/src/main/res/layout/activity_enter_mnemonic.xml (hunk 5 above)
Comment (MarcRubio) [reply app/src/main/res/layout/activity_enter_mnemonic.xml:92]: This screen is old and will be gone/redone

File: app/src/main/java/com/radixdlt/android/apps/wallet/ui/fragment/greeting/GreetingFragment.kt (hunk 1 above)
Comment (MarcRubio) [reply app/src/main/java/com/radixdlt/android/apps/wallet/ui/fragment/greeting/GreetingFragment.kt]: yeap, good point

File: app/src/main/res/layout/activity_enter_mnemonic.xml (hunk 3 above)
Comment (MarcRubio) [reply app/src/main/res/layout/activity_enter_mnemonic.xml:20]: nothing to worry, was making tweaks to left over code! This screen is an old design and will be removed eventually.

**Pull request merged**